lib/AppInfo/Application.php

```diff
@@ -101,6 +101,8 @@ public function registerFilesActivity() {
 		Util::connectHook('\OCA\Files_Trashbin\Trashbin', 'post_restore', 'OCA\Activity\FilesHooksStatic', 'fileRestore');
 		Util::connectHook('OCP\Share', 'post_shared', 'OCA\Activity\FilesHooksStatic', 'share');
 		Util::connectHook('OCP\Share', 'pre_unshare', 'OCA\Activity\FilesHooksStatic', 'unShare');
+		Util::connectHook('\OC\Files\Storage\Shared', 'fopen', 'OCA\Activity\FilesHooksStatic', 'downloadedShare');
+		Util::connectHook('\OC\Files\Storage\Shared', 'file_get_contents', 'OCA\Activity\FilesHooksStatic', 'downloadedShare');
```

Comment:
Yeah, so this is intended for federation mostly? Then I will have a look how we can do that.